mobile/ios/device/ios-core.ts

```diff
@@ -1137,11 +1137,15 @@ class GDBSignalWatcher extends stream.Writable {
 
 	public _write(packet: any, encoding: string, callback: Function) {
 		try {
-			for (let i = 0; i < packet.length - 2; i++) {
+			// The reply packages take the following form (the space in the reply templates is included for clarity)
+			// ‘S AA’  or ‘T AA n1:r1;n2:r2;...’ meaning that the program received signal number AA (a two-digit hexadecimal number)
+			for (let i = 0; i < packet.length - 3; i++) {
 				if (packet[i] === getCharacterCodePoint("$") && (packet[i + 1] === getCharacterCodePoint("T") || packet[i + 1] === getCharacterCodePoint("S"))) {
```

Comment:
Can you move the values of `$`, `T` and `S` code points to a constants outside of the `for` as currently they are evaluated on each iteration